shark/shark-android/src/main/java/shark/AndroidReferenceMatchers.kt

```diff
@@ -1074,7 +1074,8 @@ enum class AndroidReferenceMatchers {
           + " instance that has a context that is the activity."
           + " Observed here: https://github.com/square/leakcanary/issues/1#issue-74450184"
       ) {
-        manufacturer == SAMSUNG && sdkInt == 19
+        (manufacturer == SAMSUNG && sdkInt == 19) ||
+          ((manufacturer == LENOVO || manufacturer == INFINIX) && sdkInt in 31..33)
```

Comment:
I think think this is the right place, the leaktrace looks like this:

```
├─ android.content.res.ResourcesImpl class
│    Leaking: NO (a class is never leaking)
│    ↓ static ResourcesImpl.mAppContext
│                           ~~~~~~~~~~~
├─ android.app.ContextImpl instance
```

So you want `staticFieldLeak("android.content.res.ResourcesImpl", "mAppContext")` which I think is defined elsewhere in this file.